File: Graph.java
// Java program to print breadthFirstSearch traversal from a given source
// vertex. breadthFirstSearch(int s) traverses vertices reachable from s.
import java.io.*;
import java.util.*;
import java.util.HashMap;

// This class represents a directed graph using adjacency
// list representation
class Graph {
	private int V; // No. of vertices
	private LinkedList<Integer> adj[]; // Adjacency Lists

	// Constructor
	Graph(int v)
	{
		V = v;
		adj = new LinkedList[v];
		for (int i = 0; i < v; ++i)
			adj[i] = new LinkedList<Integer>();
	}

	// Function to add an edge into the graph
	void addEdge(int v, int w) { adj[v].add(w); }

   int[] getReachableNodes(int start, int distance){
      return null;
   }

 	// Driver method to
	public static void main(String args[])
	{
      BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
      try {
          String[] params = bufferedReader.readLine().replaceAll("\\s+$", "").split(" ");
          String[] firstMultipleInput = bufferedReader.readLine().replaceAll("\\s+$", "").split(" ");

          int start = Integer.parseInt(params[0]);
          int distance = Integer.parseInt(params[1]);

          int n = Integer.parseInt(firstMultipleInput[0]);
          int m = Integer.parseInt(firstMultipleInput[1]);

          Graph graph = new Graph(n);

          List<List<Integer>> edges = new ArrayList<>();

          for (int i = 0; i < m; i++){
              try {
                 String[] edge = bufferedReader.readLine().replaceAll("\\s+$", "").split(" ");
                 graph.addEdge(Integer.parseInt(edge[0]), Integer.parseInt(edge[1]));
              } catch (IOException ex) {
                  throw new RuntimeException(ex);
              }
          }
          int[] reachableNodes = graph.getReachableNodes(start, distance);
      } 
      catch (IOException ex) {
          throw new RuntimeException(ex);
      }

	}
}

File: graph.py
# Python3 Program to print breadth_first_search traversal
# from a given source vertex. breadth_first_search(int s)
# traverses vertices reachable from s.

# This class represents a directed graph
# using adjacency list representation


class Graph:

   # Constructor
   def __init__(self, num_vertices):

      # default dictionary to store graph
      self.graph = {}
      self.num_vertices = num_vertices

   # function to add an edge to graph
   def add_edge(self, u, v):
      if not u in self.graph:
         self.graph[u] = [v]
      else:
         self.graph[u].append(v)

   def get_reachable_nodes(self, start, distance):
      visited = [False] * (max(self.graph) + 1)
      queue = []
      queue.append(start)
      visited[start] = True
      count_distance = -1
      return_string = " "
      while queue:
         s = queue.pop(0)
         return_string += " " + str(s)
         for i in self.graph[s]:
            if not visited[i]:
               queue.append(i)
               visited[i] = True
            if visited[i]:
               continue
         count_distance += 1
         if count_distance <= distance:
            continue
         if count_distance > distance:
            break
      return return_string

if __name__ == '__main__':

     params = input().rstrip().split()
     start_node =  params[0];
     distance = params[1]
     start_node = int(start_node)
     distance = int(distance)
     first_multiple_input = input().rstrip().split()
     n = int(first_multiple_input[0])
     m = int(first_multiple_input[1])

     g = Graph(n)
     edges = []

     for e in range(m):
        edge = input().rstrip().split()
        res = [int(i) for i in edge]
        g.add_edge(res[0],res[1])

     reachable = g.get_reachable_nodes(start_node, distance)
     print(reachable)
